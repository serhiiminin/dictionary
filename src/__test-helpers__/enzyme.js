import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';

const shallowWithTheme = (tree, theme) => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext()
  return shallow(tree, { context })
}

const mountWithTheme = (tree, theme) => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext()

  return mount(tree, {
    context,
    childContextTypes: ThemeProvider.childContextTypes,
  })
}

export { shallowWithTheme, mountWithTheme }
