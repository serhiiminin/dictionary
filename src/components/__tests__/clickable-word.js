import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { ClickableWord } from '..';

describe('Clickable Word', () => {
  const component = renderer.create(
    <ClickableWord
      word='word'
      onClick={() => {}}
    />
  );
  const tree = component.toJSON();

  test('render', () => {
    expect(tree).toMatchSnapshot();
  });

  test('onClick', () => {
    const fn = jest.fn();
    const wrapper = shallow(
      <ClickableWord
        word='word'
        onClick={fn}
      />
    );

    wrapper.simulate('click', { preventDefault() {} });
    expect(fn).toHaveBeenCalled();
  });
});
