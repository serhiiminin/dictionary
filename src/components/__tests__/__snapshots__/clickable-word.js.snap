// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clickable Word render 1`] = `
<a
  className="ClickableWord-clickableWord-0-1-1"
  href="/"
  onClick={[Function]}
>
  word
</a>
`;
